### DISCUSSION TITLE
Winmatic HM-SEC-WIN

### DISCUSSION THREAD

**Discussion by ByteWolfGang** [SukramJ/aiohomematic#874] [Q&A] [answered]
This device has a speciality using a negative float for the closed and locked  position (-0.005) on the datapoint LEVEL.
HA represents the device as cover device with a control from 0 - 100 and misses the -0.005 value. So the window only closes/tilt, but does not lock
via UI.
I am able to set the lock value to -0.005 via an action service and the window closes and locks but it is not possible to read that state (current_position is always 0 instead of -0.005). In the HA UI the window status and history is always displayed as open what is also weird.
Anyone out here using Winmatic devices successful with this great integration?
All my Homematic devices are working well beside the Winmatic.

**Comment by SukramJ** [comment, marked as answer]
Simple answer: 
1. The value range of  for a [cover position](https://developers.home-assistant.io/docs/core/entity/cover) in HA is 0-100.
2. The state 0 means closed and open otherwise. There is no state locked

**Reply by ByteWolfGang** [reply]
This confirms the suspicion that Winmatic device integration is implemented incorrectly. I will try to create a custom cover device. Thank you!

**Reply by SukramJ** [reply]
Later releases will interpret also negative position values as closed, but state locked is not part of the HA spec.

**Reply by ByteWolfGang** [reply]
Yes you're right. I tried to play with custom cover but what we need would be a custom device feature of HA. This is already a feature request. 
Maybe you can remap a HA position of 0 to a Homematic LEVEL of -0.005 and a HA position of 1 to Homematic LEVEL 0. HA 2-100 to HM 2-100. It would be dirty but better than nothing at the moment?

**Reply by ByteWolfGang** [reply]
When I think about it...it would be correct, because HM LEVEL 0 is not a physical closed window. Its opened but not tilted. HA 1 would not be so wrong as a value.

**Reply by SukramJ** [reply]
CLOSED: level = -0.005
OPEN: level > -0.005
Moving the slider to 0 will lock the window(level=-0.005)
Moving the slider to 1 will close the window(level=0.0)

OK?

**Reply by ByteWolfGang** [reply]
Yes, correct 👍
